File: lib/python/lib/hyper.py
def hyper(m, n, r) :
    """Do the hyper level calculation.
hyper(m, 1, r) -> m + r
hyper(m, 2, r) -> m * r
hyper(m, 3, r) -> m ** r"""
    
    if not isinstance(n, int) :
        raise TypeError("argument 2 must be int, not "+str(type(n)).split("'")[1])
    if n <= 0 :
        raise ValueError("argument 2 must be a positive")
    if n == 1 :
        return m + r
    elif n == 2 :
        return m * r
    elif n == 3 :
        return m ** r
    else :
        res = 1
        for i in range(r) :
            res = hyper(m, n-1, res)
        return res

File: lib/c/sources/hyper.c
long double hyper(long double m, unsigned long n, long double r)
{
    if ( n == 1 )
    {
        return m + r ;
    }
    else if ( n == 2 )
    {
        return m * r ;
    }
    else
    {
        long double res ;
        res = 1.0 ;
        for ( unsigned long i = 0; i < r; i++ )
        {
            res = hyper(m, n-1, res) ;
        } ;
        return res ;
    } ;
} ;
